pkg/route/api/validation/validation.go

```diff
@@ -38,7 +40,7 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 		}
 	}
 
-	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, specPath.Child("wildcardPolicy")); err != nil {
+	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, route.Annotations[HostGeneratedAnnotationKey] == "true", specPath.Child("wildcardPolicy")); err != nil {
```

Comment:
where is this annotation getting set? shouldn't we be rejecting the route at that point, rather than setting a generated host in combination with a wildcard request?